File: linkList/linkListNthPoisition.c
#include<stdio.h>
#include<stdlib.h>
struct Node{
    int data;
    struct Node* next;
};
struct Node* head;
void Insert(int data, int n)
{
    struct  Node* temp = (struct Node*)malloc(sizeof(struct Node));
    temp->data = data;
    temp->next = NULL;
    if (n == 1)
    {
        temp->next = head;
        head       = temp;
        return;
    }
    struct Node* temp1 = head;
    for (int i = 0; i < n-2; i++)
    {
        temp1 = temp1->next;
    }
    temp->next = temp1->next;
    temp1->next = temp;
}
void print()
{
    struct  Node* temp = head;
    printf("List is: ");
    while (temp != NULL)
    {
        printf("%d ",temp->data);
        temp = temp->next;
    }
    printf("\n");
};
int main(){
    head = NULL;
    Insert(3,1);
    Insert(4,2);
    Insert(2,1);
    print();
}

File: linkList/linkListNthPoisition.php
<?php
class NewNode{
    public $data;
    public $next;

    public function __construct($data)
    {
        $this->data = $data;
    }
}

class LinkList{
    public $temp = null;
    public function insert(int $data = null, int $n = null)
    {
        $newNode = new NewNode($data);
        $newNode->data = $data;
        $newNode->next = null;
        if($n == 1)
        {
            $newNode->next = $this->temp;
            $this->temp    = &$newNode;
            return;
        }else{
            $currentNode = $this->temp;
            for($i = 0; $i < $n-2; $i++){
                $currentNode = $currentNode->next;
            }
            $currentNode->next = $newNode;

        }
    }
    public function display(){
        $node = $this->temp;
        while($node !== NULL){
            echo $node->data."\n";
            $node = $node->next;
        }
    }
}
$linkList = new LinkList();
$linkList->insert(20, 1);
$linkList->insert(25, 2);
$linkList->insert(25, 1);

$linkList->display();

